std/exception.d

```diff
@@ -2145,3 +2145,58 @@ pure nothrow @safe unittest
 
     auto infSlice = infinite[0 .. $]; // this would throw otherwise
 }
+
+
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructors()
```

Comment:
Please make this use camelCase rather than PascalCase. It's not a type and does not mix in a type. It mixes in functions.

Also, maybe there will be disagreement on this among the various devs, but I think that this name is _way_ too long. Something like `stdExceptionCons` seems clear enough while keeping the name from getting insanely long. I'd prefer to make it shorter, but `stdExcCons` seems like it's lost its clarity, since `Exc` isn't particularly clear. Maybe `stdExceptCons`? I'm not sure how clear that is. `stdExceptionCons` is probably better. I'd still like to come up with a shorter name than that, but I can't think of a short one that's good enough at the moment. Maybe someone else will have a better suggestion.